dpnp/fft/dpnp_utils_fft.py

```diff
@@ -423,10 +506,14 @@ def _validate_out_keyword(a, out, s, axes, c2r, r2c):
         # validate out shape against the final shape,
         # intermediate shapes may vary
         expected_shape = list(a.shape)
-        for s_i, axis in zip(s[::-1], axes[::-1]):
-            expected_shape[axis] = s_i
         if r2c:
-            expected_shape[axes[-1]] = expected_shape[axes[-1]] // 2 + 1
+            expected_shape[axes[-1]] = s[-1] // 2 + 1
+        if c2c:
+            expected_shape[axes[-1]] = s[-1]
+        for s_i, axis in zip(s[-2::-1], axes[-2::-1]):
+            expected_shape[axis] = s_i
+        if c2r:
+            expected_shape[axes[-1]] = s[-1]
```

Comment:
Isn't that already be done at line#512?

Reply:
The order in which `expected_shape` is updated important in a use case when we have repeated axis.  Line 512 is for complex-to-complex FFT and the update should start from the last axis to the first one.
This line is for complex-to-real FFT for which last axis should be updated last.